server/data/data-library/src/main/java/org/apache/james/rrt/lib/CanSendFromImpl.java

```diff
@@ -35,12 +38,18 @@
 import org.apache.james.rrt.api.RecipientRewriteTable;
 import org.apache.james.rrt.api.RecipientRewriteTableException;
 import org.apache.james.util.OptionalUtils;
+import org.apache.james.util.StreamUtils;
 
 import com.github.fge.lambdas.Throwing;
 import com.github.steveash.guavate.Guavate;
 
 public class CanSendFromImpl implements CanSendFrom {
 
+    @FunctionalInterface
+    interface DomainFetcher {
+        List<Domain> fetch(Username user);
```

Comment:
This AliasDomain relation is implicit?

Reply:
I do not get it

Reply:
I dont understand what this interface is supposed to do. Fetch domain aliases applicable for a given username?

Reply:
It is, I do not know where you want to go :/